# Eksplorasi Algoritma Id3Estimator
oleh 13520135/M. Alif Putra Yasa dan 13520165/Ghazian Tsabit Alkamil

In [8]:
# Perform ID3 algorithm on the breast cancer dataset
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
import six
import sys
sys.modules['sklearn.externals.six'] = six
from id3 import Id3Estimator
import pickle


# Load data
breast_cancer = load_breast_cancer()

# split data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(breast_cancer.data, breast_cancer.target, test_size=0.2, random_state=42)

# Create an instance of the Id3Estimator
estimator = Id3Estimator()

# Train the estimator
estimator.fit(X_train, y_train, check_input=True)

# save the model
with open('id3_model.pkl', 'wb') as f:
    pickle.dump(estimator, f)

# load the model
with open('id3_model.pkl', 'rb') as f:
    estimator = pickle.load(f)

# Predict
y_pred = estimator.predict(X_test)

# Evaluate
report = classification_report(y_test, y_pred)
print(report)

              precision    recall  f1-score   support

           0       0.95      0.88      0.92        43
           1       0.93      0.97      0.95        71

    accuracy                           0.94       114
   macro avg       0.94      0.93      0.93       114
weighted avg       0.94      0.94      0.94       114



Berdasarkan hasil eksplorasi yang telah dilakukan dapat dilihat bahwa hasil evaluasi untuk algoritma Id3Estimator memiliki nilai rata-rata sebesar 0.94 untuk metric precision, nilai rata-rata sebesar 0.93 untuk metric recall, dan nilai rata-rata sebesar 0.93 untuk metric f1.